google/oauth2/id_token.py

```diff
@@ -230,11 +237,9 @@ def fetch_id_token(request, audience):
 
                 info = json.load(f)
                 if info.get("type") == "service_account":
```

Comment:
Should we handle other credential types like `authorized_user` to keep consistent with google.autt.default()? See: https://github.com/googleapis/google-auth-library-python/blob/738611bd2914f0fd5fa8b49b65f56ef321829c85/google/auth/_default.py#L36

Reply:
For reference, you can look at Apache Airflow project: https://github.com/apache/airflow/blob/778be7993d4a8d752680d04db4edeb793386fa0c/airflow/providers/google/common/utils/id_token_credentials.py#L83-L104